mtp_cashbook/apps/mtp_auth/middleware.py

```diff
@@ -0,0 +1,23 @@
+from django.utils.functional import SimpleLazyObject
+
+from . import get_user as auth_get_user
+
+
+def get_user(request):
+    """
+    Returns a cached copy of the user if it exists or calls `auth_get_user`
+    otherwise.
+    """
+    if not hasattr(request, '_cached_user'):
+        request._cached_user = auth_get_user(request)
+    return request._cached_user
```

Comment:
Yup, fine. To make a point: we wouldn't be having this discussion if this were in a discrete commit explaining why we're rolling our own and that we're basically mirroring the Django core auth with small changes to support oauth.